### PR TITLE
components_base.dm code cleanup + docs

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Readability improvements + docs

## Why It's Good For The Game
Better code

## Changelog
:cl:
code: readability, docs
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/components_base.dm [hunk 1]
```diff
@@ -3,12 +3,15 @@
 
 /obj/machinery/atmospherics/components
 	hide = FALSE
-
-	var/welded = FALSE //Used on pumps and scrubbers
+	///Check if the component should be welded (mainly used by scrubbers and vents)
+	var/welded = FALSE
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/components_base.dm:7]: This isn't actually true, welded doesn't allow a component to be welded, it's the "has this been welded or not" var. I have no idea why it's on components as a base

File: code/modules/atmospherics/machinery/components/components_base.dm [hunk 2]
```diff
@@ -3,12 +3,15 @@
...
+	///Check if the component should show the pipe underneat the component
 	var/showpipe = TRUE
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/components_base.dm:9]:
```suggestion
	///Should the component should show the pipe underneath it?
	var/showpipe = TRUE
```

File: code/modules/atmospherics/machinery/components/components_base.dm [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
-	var/shift_underlay_only = TRUE //Layering only shifts underlay?
-
+	///When the component is in another layer other than the default one, should be shifted only the underlay pipe or the whole component?
+	var/shift_underlay_only = TRUE
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/components_base.dm:11]:
```suggestion
	///When the component is on a non default layer should we shift everything? Or just the underlay pipe
	var/shift_underlay_only = TRUE
```

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Looks pretty good, some grammatical/factual nitpicks on my end, don't take em to heart.

I think I've mentioned this before, but when you're documenting remember that what a var/proc does is less important then why it does something. Also be careful about using "Called by" in procs that aren't hyper specific, don't want to end up with an outdated comment. This pr looks just fine tho.

File: code/modules/atmospherics/machinery/components/components_base.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/atmospherics/machinery/components/components_base.dm]: should be good like this?

Comment (LemonInTheDark) [reply code/modules/atmospherics/machinery/components/components_base.dm]: Yeet

**Review by LemonInTheDark — APPROVED** [review]

**Pull request merged**